### PR TITLE
Remove activitystatus js. Add submitOnce handler for activity create

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Related to #13333 which has now been merged.

On *some* sites activities can be duplicated by clicking submit/save multiple times when adding an activity.  It does not seem easy to reproduce on other sites (eg. dmaster demo).  However, looking at the code and simulating multiple clicks in the browser it is certainly possible to submit and there is no protection against creating multiple activities.  So I suspect that slower servers/slower browsers are more likely to create duplicates.

This PR also **removes** javascript which is meant to warn the user if a scheduled activity is in the past or a completed one in the future.  However, this js doesn't work on either dmaster.demo or any of my test/live sites and has not been modified for 5 years.  As this code also adds an onclick handler it makes it much more difficult to add the "submitOnce" handler.  As it seems like this code has not worked for a long time I think it is sensible to just remove it and simplify the form - a site could implement similar functionality via jquery in an extension if desired?

Before
----------------------------------------
Possible to create multiple activities by clicking "save" multiple times.

After
----------------------------------------
Only possible to click "save" once so activity create is only submitted once.

Technical Details
----------------------------------------
Relies on #13333 which adds an onclick handler to the submit buttons.

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by jitendrapurohit** [comment]
I was able to create more than 1 activities by hitting multiple times on a save button before the form is actually submitted. So this actually replicated on  my local setup hence thought of giving it a review :-).

Tested by applying this + #13333 changes on my local - I get a series of warning after submitting the activity form.

image

**Comment by mattwire** [comment]
@jitendrapurohit Thanks for review - I can't reproduce this (no php warnings)... did you have some other customisation, clearcache? or can you provide steps to reproduce?

**Comment by jitendrapurohit** [comment]
To replicate - 

- Create an activity from civicrm/activity?reset=1&action=add&context=standalone. The form should be opened on a window and not on the ajax dialog box.
- Click save - the page redirects to the civicrm dashboard.
- Open any link from the recent items, eg the view activity link which was created from the above point.
- This should list the notices on the page. It works fine without this PR applied.

Hope that helps.

**Comment by mattwire** [comment]
@jitendrapurohit Thankyou, that should be fixed now. Can you try again?

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by mattwire** [comment]
@colemanw Thanks for merging #13333 Are you happy to merge this one too? As it follows on from that one.

File: CRM/Activity/Form/Activity.php [hunk 1]
```diff
@@ -797,25 +797,20 @@ public function buildQuickForm() {
...
-      $buttons[] = array(
-        'type' => 'cancel',
-        'name' => ts('Done'),
-      );
-      $this->addButtons($buttons);
+      $this->addButtons([
+        [
+          'type' => 'cancel',
+          'name' => ts('Done'),
+        ]
```
Comment (colemanw) [inline CRM/Activity/Form/Activity.php]: @mattwire I think technically there should be a comma here.

Comment (mattwire) [reply CRM/Activity/Form/Activity.php]: Ok, I've added a comma

**Comment by colemanw** [comment]
In my testing, I was able to create 15 activities at once before applying this patch (I'm a fast clicker).
After the patch, only one :)
Good to merge, thanks Matt.

**Comment by mattwire** [comment]
test this please

**Pull request merged**